### PR TITLE
Chrome Apps don't get window.localStorage

### LINKED ISSUE #1562: Configurator hang up since build #247
Hi, i'm using the betaflight-configurator_10.6.0_chromeos.zip-content with Chrome running at Win 7  64. No problem untill build #246. Since then the configurator hangs up showing only parts of the starting screen :
Configurator not working V247

It looks like this when working until Build #246:

Configurator working V246

Any clue ? Thanks :-)

### PR DESCRIPTION
**Pull request by Docteh**
Fixes #1562 
So, turns out Chrome Apps don't provide localStorage. In their quest to migrate people away from Apps I guess they didn't consider that people might use the same code for multiple run methods....

`window.localStorage is not available in packaged apps. Use chrome.storage.local instead.`

I think we could check like `if (!nwGui && chrome.storage && chrome.storage.local) {` if we want to use localStorage on the nwjs build.

### CODE REVIEW

File: src/js/ConfigStorage.js [hunk 1]
```diff
@@ -5,6 +5,9 @@
 var ConfigStorage = {
     // key can be one string, or array of strings
     get: function(key, callback) {
+      if (chrome.storage && chrome.storage.local) {
+          chrome.storage.local.get(key,callback);
+      } else {
```
Comment (McGiverGim) [inline src/js/ConfigStorage.js]: It must not be just the opposite? If windows.localStorage, then use the old code. If not, go back to the old chrome.storage. Or the nw builds did not have the chrome.storage?

Comment (Docteh) [reply src/js/ConfigStorage.js]: When I tried looking at window.localStorage in the Chrome APP it complained in the console. I didn't check to see if it'd complain every time I checked.

Comment (Docteh) [reply src/js/ConfigStorage.js]: Maybe its not a big problem, dev tools does collect them together...
image

Comment (mikeller) [reply src/js/ConfigStorage.js]: @Docteh:
- we already get other console error messages in the Chrome app (`history.length` is the same as yours, and it took me a lot of hacking away at the Google Analytics one to concede that this just isn't fixable;
- in your case, would checking `window.hasOwnProperty('localStorage')` be away to test without triggering the warning?

Comment (Docteh) [reply src/js/ConfigStorage.js]:
```
window.hasOwnProperty('localStorage')
true
```

Looks like we're checking once and remembering the result.

Comment (mikeller) [reply src/js/ConfigStorage.js]: @Docteh:

> window.hasOwnProperty('localStorage')
> true

That's in the Chrome app?

Comment (Docteh) [reply src/js/ConfigStorage.js]: Yeah, Chrome App on windows. I can't check Chrome App on ChromeOS.

Comment (mikeller) [reply src/js/ConfigStorage.js]: lol, they added the function, only to make it spit out an error message.

We use `getNwGui()` in `src/js/main.js` to detect NW / Chrome in a centralised place. Do you think we should use this in this case as well?

Comment (Docteh) [reply src/js/ConfigStorage.js]: I would like to find a proper way to check if we're running as a Chrome App, as running a regular web page outside of a Chrome App wouldn't get a nwGui object either.

Comment (mikeller) [reply src/js/ConfigStorage.js]: Agreed - we probably need a split between NW.js, Chrome web app, and 'other'.

Comment (Docteh) [reply src/js/ConfigStorage.js]: how about in `src/js/main.js` we check if we don't have nwGui, but have `chrome.storage.local`? We could set something to false if that is the case.

Comment (mikeller) [reply src/js/ConfigStorage.js]: Sounds reasonable - or create an 'enum' for NW / Chrome / other.

**Comment by Docteh** [comment]
Tried making an 'enum'. 

The `var nwGui` at the top of `src/js/main.js` gets passed along to the CLI tab. Didn't touch that yet, as I want to be sure I'm headed the right direction with this.

Could also do `if (GUI.Mode == GUI_Modes.ChromeApp)`

**Comment by mikeller** [comment]
Looks good to me! We should probably combine this with the checking up top in `main.js`, so that we have only one place to maintain the detection logic.

**Comment by Docteh** [comment]
The code in `src/js/tabs/cli.js` looks like it properly passes through the nwGui reference. I'm going to trust it so much that I'm going to force push and then test it :)

**Comment by Docteh** [comment]
Clipboard still works in ChromeApp and NW.js, 👍  I pulled out some stuff from `main.js`, I think I got everything relevant.

**Pull request merged**